mlrun/auth/utils.py

```diff
@@ -0,0 +1,127 @@
+# Copyright 2025 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import typing
+
+import yaml
+
+import mlrun.common.schemas
+import mlrun.errors
+import mlrun.utils.helpers
+from mlrun.config import config
+
+
+def load_secret_tokens_from_file(
+    token_file: typing.Optional[str] = None,
+    raise_on_error: bool = True,
+) -> list[dict]:
+    """
+    Load and parse secret tokens from a file.
+
+    This function reads the token file and returns the raw list of token dictionaries.
+    It does NOT validate the tokens.
+
+    :param token_file: Path to the secret tokens file. If None, uses MLRun config.
+    :param raise_on_error: Whether to raise exceptions on read/parse failure.
+    :return: List of token dictionaries from 'secretTokens'.
+    """
+    token_file = token_file or config.auth_with_oauth_token.auth_token_file
+    data = _read_secret_tokens_file(token_file, raise_on_error=raise_on_error)
+    if not data:
+        return []
+
+    tokens_list = data.get("secretTokens")
+    if not isinstance(tokens_list, list) or not tokens_list:
+        mlrun.utils.helpers.raise_or_log_error(
+            f"Invalid token file: 'secretTokens' must be a non-empty list in {token_file}",
+            raise_on_error,
+        )
+        return []
+
+    return tokens_list
+
+
+def validate_secret_tokens(
+    tokens_list: list[dict], token_file: str, raise_on_error: bool = True
+) -> list[mlrun.common.schemas.SecretToken]:
```

Comment:
it doesn't only validate, but also translates to `SecretToken`, so either change the name or move translating to separate function